Dockerfile

```diff
@@ -1,22 +1,33 @@
 FROM debian:jessie
 
-ARG DOWNLOAD_URL="https://s3-us-west-2.amazonaws.com/grafana-releases/master/grafana_latest_amd64.deb"
+ARG GRAFANA_VERSION="latest"
+ARG GF_HOME="/usr/share/grafana"
 
-RUN apt-get update && \
-    apt-get -y --no-install-recommends install libfontconfig curl ca-certificates && \
-    apt-get clean && \
-    curl ${DOWNLOAD_URL} > /tmp/grafana.deb && \
-    dpkg -i /tmp/grafana.deb && \
-    rm /tmp/grafana.deb && \
-    curl -L https://github.com/tianon/gosu/releases/download/1.10/gosu-amd64 > /usr/sbin/gosu && \
-    chmod +x /usr/sbin/gosu && \
+RUN apt-get update && apt-get install -qq -y wget tar sqlite libfontconfig curl ca-certificates && \
+    wget -O /tmp/grafana.tar.gz https://s3-us-west-2.amazonaws.com/grafana-releases/release/grafana-$GRAFANA_VERSION.linux-x64.tar.gz && \
+    tar -zxvf /tmp/grafana.tar.gz -C /tmp && rm /tmp/grafana.tar.gz && \
+    mv /tmp/grafana-* $GF_HOME && \
     apt-get autoremove -y && \
     rm -rf /var/lib/apt/lists/*
+    
+RUN mkdir -p /etc/grafana/provisioning/datasources && \
+    mkdir -p /etc/grafana/provisioning/dashboards && \
+    mkdir -p /var/lib/grafana/plugins && \
+    mkdir -p /var/log/grafana && \
+    cp $GF_HOME/conf/sample.ini /etc/grafana/grafana.ini && \
+    cp $GF_HOME/conf/ldap.toml /etc/grafana/ldap.toml && \
+    cp $GF_HOME/bin/grafana-server /usr/sbin/grafana-server && \
+    cp $GF_HOME/bin/grafana-cli /usr/sbin/grafana-cli && \
+    chown -R nobody:nogroup /var/lib/grafana && \
```

Comment:
Don't use `nobody` as a user for a service, even inside docker.   Create a specific user for this service.   You don't want someone to bind mount things and files to be owned by the user `nobody` .  `nobody` is special in unix due to some nasty legacy.

Reply:
There doesn't seem to be a good way to solve this problem when you're trying to mount directories from the host filesystem as volumes, because you need to ensure that there is a matching uid & gid between the process running in the container and on the host.

Reply:
That makes `nobody` and even worse option, since `nobody` has a _very_ explicit meaning and you should _never_ run services as nobody.

Reply:
After some discussion with @xlson the proposed solution is to explicitly set a uid and gid (exact value tbd) for the `grafana` user & group in the container, make all Grafana files world-readable in the container, and use `USER grafana:grafana`.  Users looking to mount external volumes can either `chown` them to that uid or use the `-u` docker option to run as a different uid.

If the uid and gid are specified with `ARG` it would also be relatively straightforward to build a custom image with whatever uid/gid the end user required.